src/ring_theory/graded_algebra/basic.lean

```diff
@@ -158,3 +158,58 @@ begin
 end
 
 end graded_algebra
+
+section useful_lemmas
+
+variables {ι R A : Type*}
+variables [decidable_eq ι] [add_monoid ι] [comm_semiring R] [semiring A] [algebra R A]
+variables (𝒜 : ι → submodule R A)
+
+lemma graded_algebra.nsmul_mem (n : ℕ) {i : ι} {a : A} (ha : a ∈ 𝒜 i) : n • a ∈ 𝒜 i :=
+begin
+  induction n with n ih,
+  { simp, },
+  { rw [succ_nsmul],
+    exact submodule.add_mem _ ha ih, }
+end
+
+end useful_lemmas
+
+section grade_zero
+
+variables {ι R A : Type*}
+variables [decidable_eq ι] [add_monoid ι] [comm_semiring R] [semiring A] [algebra R A]
+variables (𝒜 : ι → submodule R A)
+
+instance graded_algebra.grade_zero_semiring (𝒜 : ι → submodule R A) [graded_algebra 𝒜] :
```

Comment:
This already exists, it's `direct_sum.grade_zero.semiring`

Reply:
Thanks!